### PR TITLE
Replace dependency `crc` with `amaranth.lib.crc`

### PR DESCRIPTION
**Pull request by icb-**
Fixes GlasgowEmbedded/glasgow#371

### CODE REVIEW

**Review by whitequark — COMMENTED** [review]
Thanks! These lines are really long--could you wrap them to make them fit into 100 characters please?

File: software/glasgow/applet/memory/floppy/__init__.py [hunk 1]
```diff
@@ -1002,7 +1002,7 @@ def iter_tracks(self, file):
             size, cylinder, head = struct.unpack(">LBB", header)
             yield cylinder, head, file.read(size)
 
-    crc_mfm = staticmethod(crc.Calculator(crc.Configuration(width=16, polynomial=0x11021, init_value=0xffff)).checksum)
+    crc_mfm = staticmethod(amaranth.lib.crc.Algorithm(crc_width=16, polynomial=0x1021, initial_crc=0xffff, reflect_input=False, reflect_output=False, xor_output=0x0000)(data_width=8).compute)
```
Comment (whitequark) [inline software/glasgow/applet/memory/floppy/__init__.py:1005]: This is `amaranth.lib.crc.catalog.CRC16_CCITT_FALSE`

File: software/glasgow/applet/sensor/scd30/__init__.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def __init__(self, interface, logger):
     def _log(self, message, *args):
         self._logger.log(self._level, "SCD30: " + message, *args)
 
-    _crc = staticmethod(crc.Calculator(crc.Configuration(width=8,polynomial=0x131,init_value=0xff)).checksum)
+    _crc = staticmethod(amaranth.lib.crc.Algorithm(crc_width=8, polynomial=0x31, initial_crc=0xff, reflect_input=False, reflect_output=False, xor_output=0x00)(data_width=8).compute)
```
Comment (whitequark) [inline software/glasgow/applet/sensor/scd30/__init__.py:48]: This is `amaranth.lib.crc.catalog.CRC8_NRSC_5` for... some reason

File: software/glasgow/applet/sensor/sen5x/__init__.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def __init__(self, interface, logger):
     def _log(self, message, *args):
         self._logger.log(self._level, "SEN5x: " + message, *args)
 
-    _crc = staticmethod(crc.Calculator(crc.Configuration(width=8,polynomial=0x131,init_value=0xff)).checksum)
+    _crc = staticmethod(amaranth.lib.crc.Algorithm(crc_width=8, polynomial=0x31, initial_crc=0xff, reflect_input=False, reflect_output=False, xor_output=0x00)(data_width=8).compute)
```
Comment (whitequark) [inline software/glasgow/applet/sensor/sen5x/__init__.py:45]: This is `amaranth.lib.crc.catalog.CRC8_NRSC_5` also

File: software/glasgow/protocol/onfi.py [hunk 4]
```diff
@@ -14,7 +14,7 @@ class ONFIParameterError(Exception):
     pass
 
 
-_crc_onfi = staticmethod(crc.Calculator(crc.Configuration(width=16, polynomial=0x18005, init_value=0x4f4e)).checksum)
+_crc_onfi = staticmethod(amaranth.lib.crc.Algorithm(crc_width=16, polynomial=0x8005, initial_crc=0x4f4e, reflect_input=False, reflect_output=False, xor_output=0)(data_width=8).compute)
```
Comment (whitequark) [inline software/glasgow/protocol/onfi.py:17]: This is the only unique CRC in the codebase, interestingly!

**Review by whitequark — COMMENTED** [review]
Thanks! I realized after posting that that you could replace all but one of the CRCs with catalog ones; would you mind doing that?

File: software/glasgow/protocol/onfi.py (hunk 4 above)
Comment (whitequark) [reply software/glasgow/protocol/onfi.py]: The initial value is `b"ON"`, for "ONFI" presumably. Interesting choice.

**Comment by whitequark** [comment]
On second thought I think it's fine as it is.

File: software/glasgow/applet/sensor/scd30/__init__.py [hunk 5]
```diff
@@ -45,7 +45,7 @@ def __init__(self, interface, logger):
     def _log(self, message, *args):
         self._logger.log(self._level, "SCD30: " + message, *args)
 
-    _crc = staticmethod(crc.Calculator(crc.Configuration(width=8,polynomial=0x131,init_value=0xff)).checksum)
+    _crc = staticmethod(amaranth.lib.crc.catalog.CRC8_NRSC_5(data_width=8).compute)
```
Comment (whitequark) [inline software/glasgow/applet/sensor/scd30/__init__.py:48]: I think this will fail tests due to the extra prefix

**Review by whitequark — APPROVED** [review]
LGTM, thank you for bearing with me!

**Comment by icb-** [comment]
Sorry about that. Should be good now. I should've double checked that these weren't already defined.

**Comment by whitequark** [comment]
:tada: :tada: :partying_face: Another native dependency gone! Though `crc` is one of the better ones, it still caused some issues with Python 3.12 (I think) upgrade.

**Pull request merged**